tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +174,640 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
+ *
+ * uint64_t size = 0;
+ * uint64_t file_offset = 0;
+ * tiledb_file_get_size(ctx, file, &size);
+ * void* buffer = malloc(size);
+ * tiledb_file_export_buffer(ctx, file, bytes, &size, offset, NULL);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param bytes output buffer, alloc'ed by used.
+ * @param size size to read
+ * @param file_offset file_offset to read from
+ * @param config TileDB Config object for export settings
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_export_buffer(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    void* bytes,
+    uint64_t* size,
+    uint64_t file_offset,
+    tiledb_config_t* config);
+
+/**
+ * Export a file to the provided URI
+ *
+ * * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ *
+ * tiledb_file_export_uri(ctx, file, "s3://tiledb_bucket/some_output_file",
+ * NULL);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param output_uri output uri to save to
+ * @param config TileDB Config object for export settings
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_export_uri(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    const char* output_uri,
+    tiledb_config_t* config);
+
+/**
+ * Export a file to the opened VFS file handle
+ *
+ * * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ *
+ * tiledb_vfs_t vfs*;
+ * tiledb_vfs_alloc(ctx, &vfs);
+ * tiledb_vfs_fh_t *vfs_fh;
+ * tiledb_vfs_open(ctx, vfs, "some_file", TILEDB_VFS_WRITE, &fh);
+ * tiledb_file_export_vfs_fh(ctx, file, output, NULL);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param output output vfs file handle
+ * @param config TileDB Config object for export settings
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_export_vfs_fh(
+    tiledb_ctx_t* ctx,
+    tiledb_file_t* file,
+    tiledb_vfs_fh_t* output,
+    tiledb_config_t* config);
+
+/**
+ * Sets the starting timestamp to use when opening (and reopening) the file.
+ * This is an inclusive bound. The default value is `0`.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_set_open_timestamp_start(ctx, file, 1234);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file object.
+ * @param timestamp_start The epoch timestamp in milliseconds.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_set_open_timestamp_start(
```

Comment:
Delete. The timestamp range should be passed in the config object, as this deviates from the API we use in `tiledb_array_open`. Let's mimic the array API as closely as possible.